src/prometheus-pushgateway/deploy/prometheus-pushgateway.yaml.template

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+{% set prometheus_pushgateway_port = cluster_cfg["prometheus-pushgateway"]["port"] %}
+
+{% if cluster_cfg["alert-manager"]["use-pylon"] %}
+{% set external_url = cluster_cfg["pylon"]["uri"] %}
+{% else %}
+{% set external_url = cluster_cfg["prometheus-pushgateway"]["url"] %}
+{% endif %}
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: prometheus-pushgateway
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: prometheus-pushgateway
+  template:
+    metadata:
+      name: prometheus-pushgateway
+      labels:
+        app: prometheus-pushgateway
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/path: "/prometheus-pushgateway/metrics"
+        prometheus.io/port: "{{ prometheus_pushgateway_port }}"
+    spec:
+      hostNetwork: true
```

Comment:
Can we change hostNetwork to false?

Reply:
> Can we change hostNetwork to false?

Not sure, why change it?

Reply:
using hostNetwork break the network isolation, If we don't need to access all network interface, set hostNetwork to false will make pod more security

Reply:
Changed

Reply:
We may remove this after pylon gets configured. Currently we'll not be able to access the Pushgateway portal with hostNetwork set as true.